### PR TITLE
Add scalar function pg_get_function_result required by conn.getMetaData().getFunctions()

### PR DESCRIPTION
**Pull request by marregui**
complementing https://github.com/crate/crate/pull/10286 to resolve https://github.com/crate/crate/issues/9976

### CODE REVIEW

**Comment by mfussenegger** [comment]
Shouldn't the query mentioned in https://github.com/crate/crate/issues/9976 return proper values for `pg_get_function_result(oid)`?

With this PR I get:

```
cr> select pg_get_function_result(oid), proname from pg_proc where proname = 'trunc' limit 10;
+-----------------------------+---------+
| pg_get_function_result(oid) | proname |
+-----------------------------+---------+
|                        NULL | trunc   |
|                        NULL | trunc   |
|                        NULL | trunc   |
|                        NULL | trunc   |
|                        NULL | trunc   |
|                        NULL | trunc   |
|                        NULL | trunc   |
+-----------------------------+---------+
```

**Comment by marregui** [comment]
The pg_prog table was registering function OID's based on their name, instead of their signature.

This PR is ready for review, I have added a specific test.

File: server/src/test/java/io/crate/expression/scalar/systeminformation/PgGetFunctionResultFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public class PgGetFunctionResultFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void test_null_oid_results_in_null() {
+        assertEvaluate("pg_function_is_visible(null)", null);
+    }
+
+    @Test
+    public void test_system_function_result_type_text_representation() {
+        assertEvaluate("pg_get_function_result(0)",null);
```
Comment (kovrus) [inline server/src/test/java/io/crate/expression/scalar/systeminformation/PgGetFunctionResultFunctionTest.java:42]: can you maybe extract these two assertions into a separate test?

File: docs/general/builtins/scalar.rst [hunk 2]
```diff
@@ -2669,6 +2669,32 @@ Example:
     +------------------------+
     SELECT 1 row in set (... sec)
 
+.. _pg_get_function_result:
```
Comment (kovrus) [inline docs/general/builtins/scalar.rst:2672]: The changelog entry is missing.

File: docs/general/builtins/scalar.rst [hunk 3]
```diff
@@ -2669,6 +2669,32 @@ Example:
...
+
+``pg_get_function_result``
+--------------------------
+
+The function ``pg_get_function_result`` returns the text representation of the
+return value's data type of the function referred to by the OID.
```
Comment (kovrus) [inline docs/general/builtins/scalar.rst:2678]: `return value's data type of the function referred **to** by the OID.` is **to** obsolete here?

File: server/src/test/java/io/crate/integrationtests/UserDefinedFunctionsIntegrationTest.java [hunk 4]
```diff
@@ -299,6 +300,32 @@ public void test_pg_function_is_visible() throws Exception {
...
 
+    @Test
+    public void test_pg_get_function_result() throws Exception {
+        TypeSignature returnTypeSig = TypeSignature.parseTypeSignature("array(array(integer))");
+        String returnType = returnTypeSig.toString();
+        Signature signature = Signature
+            .builder()
+            .kind(FunctionType.SCALAR)
+            .name(new FunctionName(Schemas.DOC_SCHEMA_NAME, "make_2d_array"))
+            .argumentTypes(TypeSignature.parseTypeSignature("integer"))
```
Comment (kovrus) [inline server/src/test/java/io/crate/integrationtests/UserDefinedFunctionsIntegrationTest.java:311]: will be a better choise if there would be some kind of refactoring
```suggestion
            .argumentTypes(DataTypes.INTEGER.getTypeSignature()"))
```

**Review by kovrus — APPROVED** [review]

**Pull request merged**